### DISCUSSION TITLE
Typed useSearchParams

### DISCUSSION THREAD

**Discussion by Micnubinub** [remix-run/react-router#11124] [Proposals]
Could you add types to`useSearchParams<Type>()`

**Comment by kiliman** [comment]
I don't think so. `URLSearchParams` are user-provided. That is, they represent what is provided in the URL. I think you would be better off using a runtime validator like `zod`.

**Reply by Micnubinub** [reply]
I use typebox, its just so its nice to work with. It would be nice if i could use types to have the params be type safe 
<img width="1004" alt="image" src="https://github.com/remix-run/react-router/assets/4921356/3b38e759-2684-4d7c-b3f4-4a0df8a61bf9">

At the moment you can just setParams to whatever. I still validate, but would rather it not get to that
